src/main/java/Duke.java

```diff
@@ -1,10 +1,225 @@
+            System.out.println("        ____________________________________________");
+            for (int i = 1; i <= list.size(); i += 1) {
+                String taskType = list.get(i-1).getTaskType();
+                String markDoneStatus = list.get(i-1).isDone()?"[X]":"[ ]";
+
+                switch(taskType) {
+                case "ToDo":
+                    System.out.println("        [T]" + markDoneStatus + " " + i + ". " + list.get(i-1).getTaskName());
+                    break;
+
+                case "Deadline":
+                    String deadline = list.get(i-1).getTaskDeadline();
+                    System.out.println("        [D]" + markDoneStatus + " " + i + ". " + list.get(i-1).getTaskName() +
+                            " (by: " + deadline + ")");
+                    break;
+
+                case "Event":
+                    String eventTime = list.get(i-1).getEventTime();
+                    System.out.println("        [E]" + markDoneStatus + " " + i + ". " + list.get(i-1).getTaskName() +
+                            " (at: " + eventTime + ")");
+                    break;
+
+                default:
+                    break;
+                }
+            }
+            System.out.println("        ____________________________________________");
+    }
+
+    private static void printGoodbyeMessage() {
+        System.out.println("____________________________________________");
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("____________________________________________");
+    }
+
+    private static void printWelcomeMessage() {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("____________________________________________");
+        System.out.println("Hello! I'm Duke.");
+        System.out.println("What can I do for you?");
+        System.out.println("____________________________________________");
+    }
+
+    public static void handleInput() {
+        String input = "";
+        Scanner in = new Scanner(System.in);
+        input = in.nextLine();
+        String[] line = input.split(" ");
+        boolean checkedInitialLine = false;
+
+        while (!line[0].equals("bye")) {
+            if (checkedInitialLine) {
+                input = in.nextLine();
+                line = input.split(" ");
+            } else {
+                checkedInitialLine = true;
+            }
+
+            switch (line[0]) {
+            case "list":
+                printList(list);
+                break;
+
+            case "mark":
+                try {
+                    if ((Integer.parseInt(line[1])) > list.size() || (Integer.parseInt(line[1])) < 1) {
```

Comment:
Maybe you could split this up into booleans? It may be difficult to understand what is happening here.

https://nus-cs2113-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions